src/main/java/duke/relation/PatientTaskList.java

```diff
@@ -74,7 +74,21 @@ public void deletePatientTaskByUniqueId(int uid) throws DukeException {
                 return;
             }
         }
-        throw new DukeException("Such Task does not exist!");
+    }
+
+    /**
+     * .
+     *
+     * @param id .
+     * @throws DukeException .
+     */
+    public void deleteAllPatientTaskByTaskId(int id) throws DukeException {
+        for (PatientTask patientTask : patientTaskIdMap.values()) {
```

Comment:
This probably can't be helped at the moment, but listing out a HashMap's values is a huge cost in regards to efficiency. If we aim to do any optimizing in the future, we should find a more efficient way of doing this.

Reply:
noted